benchmarks/distributed/ddp/compare_ddp_sample.md

```diff
@@ -0,0 +1,23 @@
+Sample Output for compare_ddp.py
+
+Key takeaway:
+- forward performance is almost the same. DDPCore is the slowest.
+- async PythonDDP is slightly slower than DDPCore. DDP_CPP_CORE > PYTHON_DDP_ASYNC > PYTHON_DDP_SYNC > LEGACY_DISTRIBUTED_DATA_PARALLEL
+
+```
+Metrics for GPU 0 ddp_option DDPOption.DDP_CPP_CORE:
```

Comment:
Hey @bowangbj, thanks a lot for putting this comprehensive measurements together within a very short period of time! And the result looks very promising! Shall we try varying the bucket size for both C++ DDP and Python DDP and see how that affects the delay? In theory, this shouldn't make too much difference, as it does not affects the number of C++/Python context switches. But it will be helpful to know that for sure. :)

Reply:
Great suggestion! 

Added the result per different bucket size. Overall Python_DDP is 2-20% slower than DDPCore for different bucket size. As expected, larger bucket size can reduce parallelization thus it introduces small delay as bucket size increases.  Also once the bucket size is quite large, it won't make difference as bucket size continues to increase.

Please see details at sample_output.md